src/pages/Account/TokenIcon.tsx

```diff
@@ -0,0 +1,71 @@
+import { ChainId, DXD, SWPR } from '@swapr/sdk'
+
+import { Box } from 'rebass'
+
+import SWPRLogo from '../../assets/images/swpr-logo.png'
+import DXDLogo from '../../assets/svg/dxd.svg'
+import { getTokenLogoURL, NATIVE_CURRENCY_LOGO } from '../../components/CurrencyLogo/CurrencyLogo.utils'
+import { useListsByToken } from '../../state/lists/hooks'
+import { StyledLogo } from './Account.styles'
+import { getTokenURLWithNetwork } from './accountUtils'
+
+interface TokenIconProps {
+  symbol: string
+  address?: string
+  chainId?: ChainId
+  width?: number
+  height?: number
+  marginRight?: number
+}
+export function TokenIcon({ symbol, address, chainId, width = 32, height = 32, marginRight = 6 }: TokenIconProps) {
+  const allTokens = useListsByToken()
+  const token = allTokens.get(symbol)
+  let sources: string[] = []
+
+  if (address === '0x0000000000000000000000000000000000000000') {
+    sources.push(NATIVE_CURRENCY_LOGO[chainId ?? ChainId.MAINNET])
+  }
+
+  if (chainId && DXD[chainId] && DXD[chainId].address === address) {
+    sources = [DXDLogo]
+  }
+
+  if (chainId && SWPR[chainId] && SWPR[chainId].address === address) {
```

Comment:
We can take advantage of Optional Chaining Operator to save us some characters 😄

Reply:
Done 👍🏼